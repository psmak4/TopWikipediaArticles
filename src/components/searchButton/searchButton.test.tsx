import { render, screen } from '@testing-library/react'
import SearchButton from './searchButton'

describe('SearchButton component', () => {
	test('renders the component', () => {
		render(<SearchButton disabled={false} />)
		const buttonElement = screen.getByRole('button')
		expect(buttonElement).toBeInTheDocument()
	})

	test('renders with disabled attribute when disabled is true', () => {
		render(<SearchButton disabled={true} />)
		const buttonElement = screen.getByRole('button')
		expect(buttonElement).toBeDisabled()
	})

	test('renders without disabled attribute when disabled is false', () => {
		render(<SearchButton disabled={false} />)
		const buttonElement = screen.getByRole('button')
		expect(buttonElement).not.toBeDisabled()
	})

	test('renders the correct text', () => {
		render(<SearchButton disabled={false} />)
		const textElement = screen.getByText('Search')
		expect(textElement).toBeInTheDocument()
	})

	test('matches snapshot', () => {
		const { asFragment } = render(<SearchButton disabled={false} />)
		expect(asFragment()).toMatchSnapshot()
	})
})
