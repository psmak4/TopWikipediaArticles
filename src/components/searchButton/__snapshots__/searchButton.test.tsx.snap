// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="searchButton"
    type="submit"
  >
    <span
      class="text"
    >
      Search
    </span>
  </button>
</DocumentFragment>
`;
